plugins/modules/zabbix_usergroup.py

```diff
@@ -0,0 +1,518 @@
+                self._module.exit_json(changed=True)
+            return self._zapi.usergroup.update(kwargs)
+        except Exception as e:
+            self._module.fail_json(msg='Failed to update user group "%s": %s' % (kwargs['usrgrpid'], e))
+
+    def add(self, **kwargs):
+        """Add user group.
+
+        Parameters:
+            **kwargs: Arbitrary keyword parameters.
+
+        Returns:
+            usergroup: added user group
+        """
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            parameters = self._construct_parameters(**kwargs)
+            usergroup = self._zapi.usergroup.create(parameters)
+            return usergroup['usrgrpids'][0]
+        except Exception as e:
+            self._module.fail_json(msg='Failed to create user group "%s": %s' % (kwargs['name'], e))
+
+    def delete(self, usrgrpid):
+        """Delete user group.
+
+        Parameters:
+            usrgrpid: User group id.
+
+        Returns:
+            usergroup: deleted user group
+        """
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            else:
+                return self._zapi.usergroup.delete([usrgrpid])
+        except Exception as e:
+            self._module.fail_json(msg='Failed to delete user group "%s": %s' % (usrgrpid, e))
+
+
+def main():
+    argument_spec = zabbix_utils.zabbix_common_argument_spec()
+    argument_spec.update(
+        name=dict(type='str', required=True, aliases=['user_group']),
+        gui_access=dict(type='str', required=False, default='default', choices=['default', 'internal', 'LDAP', 'disable']),
+        debug_mode=dict(type='str', required=False, default='disabled', choices=['disabled', 'enabled']),
+        status=dict(type='str', required=False, default='enabled', choices=['enabled', 'disabled']),
+        rights=dict(type='list', elements='dict', required=False, options=dict(
+            host_group=dict(type='str', required=True),
+            permission=dict(type='str', required=True, choices=['denied', 'read-only', 'read-write'])
+        )),
+        tag_filters=dict(type='list', elements='dict', required=False, options=dict(
+            host_group=dict(type='str', required=True),
+            tag=dict(type='str', default=''),
+            value=dict(type='str', default='')
+        )),
+        state=dict(type='str', default='present', choices=['present', 'absent'])
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    server_url = module.params['server_url']
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    http_login_user = module.params['http_login_user']
+    http_login_password = module.params['http_login_password']
+    validate_certs = module.params['validate_certs']
+    timeout = module.params['timeout']
```

Comment:
Thank you @tcraxs for new module :)
I think these lines are unnecessary, what do you think?

Reply:
damn, good catch. I am so used to them that it didn't occurred to me they are redundant already. Btw, shouldn't some pylint catch that as a unused variable? `ansible-test sanity` is using one if I am not mistaken

Reply:
Yes, you are right. I was unsure but of course they are used in the module_utils/wrapper stuff :) Thanks a lot 👍

Reply:
> damn, good catch. I am so used to them that it didn't occurred to me they are redundant already. Btw, shouldn't some pylint catch that as a unused variable? ansible-test sanity is using one if I am not mistaken

It looks like`ansible-test sanity` doesn't allow for the detection of unused variables.
I think you'll need to use other testing tools like flake8 to detect unused variables